fairlearn/datasets/_fetch_diabetes_hospital.py

```diff
@@ -91,11 +98,21 @@ def fetch_diabetes_hospital(
     if not data_home:
         data_home = pathlib.Path().home() / _DOWNLOAD_DIRECTORY_NAME
 
-    return fetch_openml(
+    data_dict = fetch_openml(
         data_id=43874,
         data_home=data_home,
         cache=cache,
         as_frame=True,
```

Comment:
Can you use `as_frame=as_frame` to get the expected behavior without all of the ifs below?

Reply:
(What a can of worms... sorry about this... I'm just realizing through this issue as well as #1165 that we have a non-trivial dependence on OpenML.)